src/test/java/tests/BaseTest.java

```diff
@@ -1,19 +1,33 @@
 package tests;
 
+import com.github.javafaker.Faker;
 import drivers.BrowserType;
 import drivers.DriverFactory;
 import drivers.DriverManager;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.testng.annotations.AfterMethod;
 import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.BeforeTest;
+import properties.ReadProperties;
 import static drivers.DriverManager.getDriver;
 
 public class BaseTest {
+    protected final Logger logger = LogManager.getLogger(this);
+    protected Faker faker = new Faker();
+    protected ReadProperties props;
+
+    @BeforeTest
+    public void setupProperties() {
```

Comment:
do you need to set up properties before each test or it's enough to set up them before the whole test run?

Reply:
Why before each test? As far as I know, BeforeTest is called once before a number of tests regardless of how many of them are annotated.